### DISCUSSION TITLE
Formatting duration options

### DISCUSSION THREAD

**Discussion by kossnocorp** [js-fns/date-fns#3693] [Ideas]
@ckcherry23 is working on a duration improvements proposal, and one of the ideas she had is to add the ability to format duration using format string. I.e.:

```
10y 2m
23:08.456
About 8 hours
```

While I can't see clearly how it would work, given the duration has variable units, I think it's worth considering. One of the alternatives is to have a predefined set of formats like `short`, `time`, `distance`, etc.

Please share your feedback here.

**Comment by ckcherry23** [comment]
Depending on how format duration is implemented (or improved), we could also add the ability to parse strings to return a `Duration`. 

For example,
```ts
var result = parseDuration(durationString: "2 days 3 hours")
//=> { days: 2, hours: 3 }
```

```ts
var result = parseDuration(durationString: "10y 2m", formatString: "y'y' M'm'")
//=> { years: 10, months: 2 }
```

**Reply by kossnocorp** [reply]
I see, but what about if the duration is `{ days: 5374 }`? Do you expect to normalize it into those units (`y` and `m`)?

**Reply by ckcherry23** [reply]
I think we can take another argument `options` to let the user specify if they want normalization or not.
```ts
var result = parseDuration(durationString: string, formatString: string, options: ParseDurationOptions)

interface ParseDurationOptions {
  normalize?: boolean;
  //...
}
```

I think no normalization must be the default.
